src/main/java/edu/wisc/my/personalizedredirection/service/RedirectionServiceImpl.java

```diff
@@ -19,15 +18,15 @@
 public class RedirectionServiceImpl implements IRedirectionService {
 	protected final Logger logger = LoggerFactory.getLogger(getClass());
 
-	public String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws PersonalizedRedirectionException {
+	public String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws RuntimeException {
 
 		// Find search key in the header
 		String attributeToSearchFor = dataSource.getAttributeName();
 		String toFind = request.getHeader(attributeToSearchFor);
 
 		if (toFind == null || toFind.length() == 0) {
 			logger.error("Attribute " + attributeToSearchFor + " not found");
-			throw new PersonalizedRedirectionException("Attribute " + attributeToSearchFor + " not found.");
+			throw new RuntimeException("Attribute " + attributeToSearchFor + " not found.");
```

Comment:
I'm thinking of this service as a standalone with one job. You pass me your shib credentials and the name of an app, and I'll send you to another URL. 

If I don't do that job, something went wrong, so in my thinking, any deviation from the expected result is an exception, whether your ID isn't in the file, or whether the file system goes down.